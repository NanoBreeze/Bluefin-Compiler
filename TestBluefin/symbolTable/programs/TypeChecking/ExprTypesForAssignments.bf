struct First {
	float a;
};

struct Second {
	First one;
};

First f() {}


int main() {
	First first = f();
	
	float x = 5+first.a;
	
	Second second;
	int y = second.one.a;
}