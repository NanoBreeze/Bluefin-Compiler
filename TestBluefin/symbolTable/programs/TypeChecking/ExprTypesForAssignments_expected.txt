f - t_First - t_First
f() - t_First - t_First
5 - t_int - t_float
first - t_First - t_First
first.a - t_float - t_float
5+first.a - t_float - t_float
second - t_Second - t_Second
second.one - t_First - t_First
second.one.a - t_float - t_float
